rgw/v2/tests/s3_swift/reusables/sts.py

```diff
@@ -112,3 +154,377 @@ def assume_role(sts_client, **kwargs):
     assume_role_response = sts_client.assume_role(**kwargs)
     log.info(f"assume_role_response:\n{assume_role_response}")
     return assume_role_response
+
+
+class Keycloak:
+    def __init__(
+        self,
+        client_id="sts_client",
+        client_secret="client_secret1",
+        ip_addr="localhost",
+        attributes=None,
+    ):
+        """
+        Keycloak deployment and administration through curl
+        """
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.ip_addr = ip_addr
+        out = utils.exec_shell_cmd("sudo podman ps")
+        if "keycloak" in out:
+            log.info("Keycloak is already running. skipping deployment..")
+            return
+        self.install_keycloak()
+        out = utils.exec_shell_cmd("sudo yum install -y jq")
+        if out is False:
+            raise Exception("jq installation failed")
+        self.create_client()
+        self.add_service_account_roles_to_client(client_name=self.client_id)
+        self.set_audience_in_token(
+            self.client_id, "set_audience_scope", "set_audience_protocol_mapper"
+        )
+        self.set_session_tags_in_token(self.client_id)
+        self.realm_keys_workaround()
+        if attributes:
+            self.add_user_attributes(attributes=attributes, username="admin")
+
+    def install_keycloak(self):
+        out = utils.exec_shell_cmd(
+            "sudo podman run -d --name keycloak -p 8180:8180 -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=admin quay.io/keycloak/keycloak:22.0.0-0 start-dev --http-port 8180"
+        )
+        if out is False:
+            raise Exception("keycloak deployment failed")
+        log.info("sleeping for 60 seconds")
+        time.sleep(60)
```

Comment:
Do we need sleep here, as we just returning the output response after sleep

Reply:
yes @anrao19 , we need to allow some time for the keycloak server to be up